docs/tutorial/hyperfoil.adoc

```diff
@@ -1,129 +1,156 @@
 roles:
-  example:
+  setup:
     hosts:
     - test
+    setup-scripts:
+    - setup-getting-started
+    run-scripts:
+    - run-getting-started
+  hyperfoil:
+    hosts:
+    - test
+    setup-scripts:
+    - setup-hyperfoil
     run-scripts:
     - test-endpoint
-    - getting-started
+    cleanup-scripts:
+    - remove-hyperfoil
+EOF
+```
+
+Run with and replace `targethost` value with your needs
+```
+targethost=$USER@localhost
+java -jar qDup-0.8.5-uber.jar -i ~/.ssh/qdup hyperfoil.yaml -ix -S target-host=$targethost
 ```
 
 == Splitting the roles
-The previous tutorial created an `example` role to run all of ours scripts. This works but a proper performance test will normally isolate the load generation (currently `curl`) from the process we are testing (currently `quarkus dev`).
-qDup isolates scripts by assigning them to roles that are running on different `hosts`. To isolate `test-endoing` from `getting-started` we need to introduce a second role. Let us call it `hyperfoil`. The `roles` section will now look like the following:
+The previous tutorial created a role to run all of ours scripts. This works but a proper performance test will normally isolate the load generator (`hyperfoil`) from the process we are testing (`quarkus dev`).
+qDup isolates scripts by assigning them to roles that are running on different `hosts`. To isolate `test-endpoint` from `run-getting-started` we need to introduce a second role. Let us call it `hyperfoil`. The `roles` section will now look like the following:
 
 ```yaml
-# ...
 roles:
-  hyperfoil:
+  setup:
     hosts:
-    - test #we still only have 1 ssh computer
+    - test
+    setup-scripts:
+    - setup-getting-started
     run-scripts:
-    - test-endpoint
-  example:
+    - run-getting-started
+  hyperfoil:
     hosts:
     - test
+    setup-scripts:
+    - setup-hyperfoil
     run-scripts:
-    - getting-started
+    - test-endpoint
+    cleanup-scripts:
+    - remove-hyperfoil
 ```
 NOTE: We are only running an ssh server on one computer so we still use the `test` host for both roles but now we could split them if needed.
 
-=== Hyperfoil
-We named the new role `hyperfoil` but right now it is running `curl`. Open up the link:https://hyperfoil.io/quickstart/quickstart1.html[hyperfoil quickstart] ad we are going to follow it's steps in a terminal and add them to the `test-endpoint` script.
+=== Hyperfoil role
+We named the new role `hyperfoil`.
 
-NOTE: we test commands in a terminal while writing our qDup scripts to check their expected output and look for the common challenges that come with automating shell commands.
-
-The first command is to `wget` the hyperfoil release but we are going to start with `cd /tmp` like we did in `getting-started` so that we don't add files to the current home directory.
+==== setup-hyperfoil
+It will download, install Hyperfoil locally and configure the `single-request.hf.yaml` file to use `http://localhost:8080` instead of `http://hyperfoil.io`
```

Comment:
The first sentence after a new heading should probabably use a proper noun as the topic of the first sentence instead of a pronoun. Starting with `it` will force a reader to go back to the previous section to identify the topic of the new section.

Reply:
Done